packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx

```diff
@@ -0,0 +1,122 @@
+import { Typography } from "@mui/material";
+import {
+  ChangeEventHandler,
+  FocusEventHandler,
+  KeyboardEventHandler,
+  useRef,
+  FunctionComponent,
+} from "react";
+import { useBlockProtocolUpdateEntity } from "../../../components/hooks/blockProtocolFunctions/useBlockProtocolUpdateEntity";
+import { rewriteEntityIdentifier } from "../../../lib/entities";
+import { usePageContext } from "../PageContext";
+import { cleanUpTitle, focusEditorBeginning, isValidPageTitle } from "./utils";
+
+type PageTitleProps = {
+  accountId: string;
+  entityId: string;
+  value: string;
+};
+
+// TODO: Add read-only mode based on page permissions
+export const PageTitle: FunctionComponent<PageTitleProps> = ({
+  accountId,
+  entityId,
+  value,
+}) => {
+  // TODO: Display update error once expected UX is discussed
+  const { updateEntity, updateEntityLoading } =
+    useBlockProtocolUpdateEntity(true);
+  const { editorView } = usePageContext();
+
+  const titleValueRef = useRef(value);
+
+  const handleInputChange: ChangeEventHandler<HTMLHeadingElement> = (event) => {
+    const { currentTarget } = event;
+    titleValueRef.current = currentTarget.textContent || "";
+
+    /**
+     * this prevents two issues
+     * 1 - when user pastes styled stuff into title
+     * other way of achieving this is implementing https://htmldom.dev/paste-as-plain-text/
+     * this looks like working well for now on all browser.
+     * we can switch to paste-as-plain-text later if we decide this solution is not enough
+     * 2 - firefox & safari renders a `br` tag when user clears the editable area,
+     * which prevents css selector `:empty` from working. This removes the `br` as well
+     */
+    if (currentTarget?.children.length) {
+      currentTarget.innerText = currentTarget.textContent || "";
+    }
+  };
+
+  const handleInputKeyDown: KeyboardEventHandler<HTMLInputElement> = (
+    event,
+  ) => {
+    if (event.key === "Enter" || event.key === "Escape") {
+      event.currentTarget.blur();
+    }
+
+    if (event.key === "ArrowDown") {
+      const { anchorOffset, type } = window.getSelection() || {};
+
+      const isCaret = type === "Caret";
+      const isCaretAtEnd = anchorOffset === titleValueRef?.current.length;
+
+      if (isCaret && isCaretAtEnd) {
+        focusEditorBeginning(editorView);
+      }
+    }
+  };
+
+  const handleInputBlur: FocusEventHandler<HTMLInputElement> = () => {
+    const valueToSave = cleanUpTitle(titleValueRef.current);
+    if (valueToSave === value) {
+      return;
+    }
+
+    if (!isValidPageTitle(valueToSave)) {
+      titleValueRef.current = value;
```

Comment:
This creates strange UX because the displayed value doesn't revert, it just silently doesn't save.

Reply:
This is fixed here https://github.com/hashintel/hash/pull/956/commits/2d4c368e07f82739ce9328a543e1490bf442c73e 
Now users can save empty page titles as well.